DOC Small documentation of multiclass classification (#943)
fairlearn/fairlearn: docs/user_guide/assessment.rst

    >>> print("ratio in recall = ", grouped_metric.ratio(method='between_groups'))    
    ratio in recall =  0.0

> We may also be interested in multiclass classification, however, typical group fairness metrics such as equalized odds and demographic parity are only defined for binary classification. We could, for every group, define one-to-one or one-to-rest classifications and calculate such metrics on this instead. However, we can also use metrics that are defined for multiclass classification. For example, accuracy is a multiclass metric that we can utilize through scikit-learn's :py:func:`sklearn.metrics.accuracy_score` in combination with a :class:`MetricFrame` as follows:

Reviewer: ```suggestion
We may also be interested in multiclass classification. However, typical group fairness metrics such as equalized odds and demographic parity are only defined for binary classification. One way to measure fairness in the multiclass scenario is to define one-to-one or one-to-rest classifications for each group and calculate the metrics on this instead. Alternatively, we can use predefined metrics for multiclass classification. For example, accuracy is a multiclass metric that we can utilize through scikit-learn's :py:func:`sklearn.metrics.accuracy_score` in combination with a :class:`MetricFrame` as follows:
```

    >>> print("ratio in recall = ", grouped_metric.ratio(method='between_groups'))    
    ratio in recall =  0.0

> We may also be interested in multiclass classification, however, typical group fairness metrics such as equalized odds and demographic parity are only defined for binary classification. We could, for every group, define one-to-one or one-to-rest classifications and calculate such metrics on this instead. However, we can also use metrics that are defined for multiclass classification. For example, accuracy is a multiclass metric, and we can easily use this using scikit-learn's :class:`sklearn.metrics.accuracy_score` in combination with a :class:`MetricFrame` as follows:

Reviewer 2: ```suggestion
We may also be interested in multiclass classification, however, typical group fairness metrics such as equalized odds and demographic parity are only defined for binary classification. We could, for every group, define one-to-one or one-to-rest classifications and calculate such metrics on this instead. However, we can also use metrics that are defined for multiclass classification. For example, accuracy is a multiclass metric that we can utilize through scikit-learn's :class:`sklearn.metrics.accuracy_score` in combination with a :class:`MetricFrame` as follows:
```

.. doctest:: assessment_metrics

    >>> from sklearn.metrics import accuracy_score
    >>> from fairlearn.metrics import MetricFrame
    >>> # Multiclass labels
    >>> y_mult_true = [0,1,2,1,3,0,1,3,0,2,1,2,0,0,1,3]
    >>> y_mult_pred = [0,1,1,2,3,0,1,0,0,2,1,2,3,0,0,2]

>     >>> # compute metric frame

Reviewer 2: This comment doesn't really add anything on top of the following line. I'd leave it out.

>     >>> mf = MetricFrame(metric=accuracy_score,
>     ...                 y_true=y_mult_true, y_pred=y_mult_pred,
>     ...                 sensitive_features=group_membership_data)
>     >>> # print results

Reviewer 2: ```suggestion
    >>> mf = MetricFrame(metric=accuracy_score,
    ...                  y_true=y_mult_true, y_pred=y_mult_pred,
    ...                  sensitive_features=group_membership_data)
    >>> # print results
```

I think this needs more indentation (?)

    >>> print(mf.by_group) # series with accuracy for each sensitive group
    sensitive_feature_0
    a         1.0
    b         0.5
    c    0.428571
    d         1.0
    Name: accuracy_score, dtype: object

>     >>> print(mf.difference()) # difference in accuracy between the two sensitive groups

Reviewer 2: ```suggestion
    >>> print(mf.difference()) # difference in accuracy between the tmin and max of all groups
```

Author: Haha oops yes.

Author: is tmin a typo? :)

Reviewer 2: Yes! Fixed.

    >>> print("ratio in recall = ", grouped_metric.ratio(method='between_groups'))    
    ratio in recall =  0.0

> Multiclass Metrics

Reviewer 2: ```suggestion
Multiclass metrics
```

scenario is to define one-to-one or one-to-rest classifications for each group
and calculate the metrics on this instead. Alternatively, we can use predefined
metrics for multiclass classification. For example, accuracy is a multiclass
metric that we can utilize through scikit-learn's :py:func:`sklearn.metrics.accuracy_score`
in combination with a :code:`MetricFrame` as follows:

.. doctest:: assessment_metrics

    >>> from sklearn.metrics import accuracy_score
    >>> from fairlearn.metrics import MetricFrame
    >>> y_mult_true = [0,1,2,1,3,0,1,3,0,2,1,2,0,0,1,3]
    >>> y_mult_pred = [0,1,1,2,3,0,1,0,0,2,1,2,3,0,0,2]
    >>> mf = MetricFrame(metric=accuracy_score,
    ...                  y_true=y_mult_true, y_pred=y_mult_pred,
    ...                  sensitive_features=group_membership_data)
    >>> print(mf.by_group) # series with accuracy for each sensitive group
    sensitive_feature_0
    a         1.0
    b         0.5
    c    0.428571
    d         1.0
    Name: accuracy_score, dtype: object
    >>> print(mf.difference()) # difference in accuracy between the max and min of all groups
    0.5714285714285714

> Multiple Metrics in one :code:`MetricFrame`

Reviewer 2: ```suggestion
Multiple metrics in one :code:`MetricFrame`
```

dictionaries, with the first key corresponding matching that in the dictionary holding
the desired underlying metric functions.

> Non-sample Parameters

Reviewer 2: ```suggestion
Non-sample parameters
```

    d    0...
    Name: metric, dtype: object

> Multiple Sensitive Features

Reviewer 2: ```suggestion
Multiple sensitive features
```
